test/test_alias.cpp

```diff
@@ -285,4 +285,85 @@ TEST_F(AliasTest, ViewPermute) {
       __FILE__);
 }
 
+TEST_F(AliasTest, SliceViewPermute) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  constexpr int batches = 16;
+  constexpr int seq_length = 128;
+  constexpr int features = 1024;
+  constexpr int heads = 16;
+
+  // The input tensor is a concatenation of [query, key, value], and therefore
+  // has a feature dimension of size `features * 3`.
+  TensorView* in =
+      makeContigConcreteTensor({batches, seq_length, features * 3});
+  fusion->addInput(in);
+  std::vector<TensorView*> splits({
+      slice(in, {0, 0, 0}, {batches, seq_length, features}),
+      slice(in, {0, 0, features}, {batches, seq_length, features * 2}),
+      slice(in, {0, 0, features * 2}, {batches, seq_length, features * 3}),
+  });
+  for (TensorView* split : splits) {
+    split = reshape(
+        split,
+        {batches, seq_length, features},
+        {batches, seq_length, heads, features / heads});
+    split = permute(split, {0, 2, 1, 3});
+    fusion->addOutput(split);
+  }
+
+  FusionExecutorCache fec(std::move(fusion));
+  at::Tensor in_tensor = at::randn({batches, seq_length, features * 3}).cuda();
+  std::vector<at::Tensor> out_tensors = fec.runFusionWithInputs({in_tensor});
+  EXPECT_EQ(out_tensors.size(), 3);
+
+  for (const auto& out_tensor : out_tensors) {
+    EXPECT_TRUE(out_tensor.is_alias_of(in_tensor));
+  }
+
+  std::vector<at::Tensor> expected_out_tensors =
+      in_tensor.split(/*split_size=*/features, /*dim=*/-1);
+  for (auto& expected_out_tensor : expected_out_tensors) {
+    expected_out_tensor =
+        expected_out_tensor.view({batches, seq_length, heads, -1})
+            .permute({0, 2, 1, 3});
+  }
+
+  testValidate(
+      fec.fusion(),
+      out_tensors,
+      {in_tensor},
+      expected_out_tensors,
+      __LINE__,
+      __FILE__);
+}
+
+TEST_F(AliasTest, NotAllOutputsAlias) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({2, 3});
+  TensorView* slice_out = slice(in, {0, 0}, {2, 2});
```

Comment:
nitpick, can we try a static trivial slice with a view to merge the sliced dimension?

Reply:
Done.